components/script/fetch.rs

```diff
@@ -236,14 +240,25 @@ impl FetchResponseListener for FetchContext {
                         fill_headers_with_metadata(self.response_object.root(), m);
                         self.response_object.root().set_type(DOMResponseType::Cors);
                     },
-                    FilteredMetadata::Opaque => self
-                        .response_object
-                        .root()
-                        .set_type(DOMResponseType::Opaque),
-                    FilteredMetadata::OpaqueRedirect => self
-                        .response_object
-                        .root()
-                        .set_type(DOMResponseType::Opaqueredirect),
+                    FilteredMetadata::Opaque => {
+                        self.response_object
+                            .root()
+                            .set_type(DOMResponseType::Opaque);
+                        self.response_object.root().set_url_list(vec![]);
+                        self.response_object.root().set_status(None);
+                        self.response_object.root().set_raw_status(None);
+                        self.response_object.root().set_headers(None);
+                        self.response_object.root().set_body(vec![]);
+                    },
+                    FilteredMetadata::OpaqueRedirect => {
+                        self.response_object
+                            .root()
+                            .set_type(DOMResponseType::Opaqueredirect);
+                        self.response_object.root().set_status(None);
+                        self.response_object.root().set_raw_status(None);
+                        self.response_object.root().set_headers(None);
+                        self.response_object.root().set_body(vec![]);
```

Comment:
Since all of these match blocks call the same settings, why don't we move this behaviour into `set_type` instead, which then removes the need to expose the new setter functions?

Reply:
I thought so. Now, refactor such that these members are set  when you set the type?

Reply:
Yes please!

Reply:
Made a dirty fix. May have scope for refactoring. Just want your opinion before that.